packages/flutter_tools/test/general.shard/build_system/targets/web_test.dart

```diff
@@ -464,11 +464,17 @@ void main() {
   }));
 
   test('Generated service worker correctly inlines file hashes', () {
-    final String result = generateServiceWorker(<String, String>{'/foo': 'abcd'});
+    final String result = generateServiceWorker(<String, String>{'/foo': 'abcd'}, <String>[]);
 
     expect(result, contains('{\n  "/foo": "abcd"\n};'));
   });
 
+  test('Generated service worker includes core files', () {
+    final String result = generateServiceWorker(<String, String>{'/foo': 'abcd'}, <String>['foo', 'bar']);
+
+    expect(result, contains('"foo",\n"bar"'));
```

Comment:
We should also have a test that checks that we don't include stuff we don't want to include.

Reply:
Done